### PR TITLE
(refactor)operator:Added structure for arguments in engineRunnerPod AND engineMonitorService

### PR DESCRIPTION
**Pull request by amitbhatt818**
Signed-off-by: Amit Bhatt <amitbhatt818@gmail.com>

**What this PR does / why we need it**:

- Refactor the engineRunnerPod and engineMonitorService function to improve code quality and BCH score.

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 1]
```diff
@@ -192,14 +189,31 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
...
+		pod:             &corev1.Pod{},
+		engineRunner:    engineRunner,
+		reconcileEngine: engineReconcile,
+	}
+
+	monitorService := &serviceEngineMonitor{
+		service:         &corev1.Service{},
+		engineMonitor:   engineMonitor,
+		reconcileEngine: engineReconcile,
+	}
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]: Can you define these variables in types.go file

File: pkg/controller/chaosengine/types.go [hunk 2]
```diff
@@ -51,3 +53,21 @@ type applicationInfo struct {
...
+// reconcileEngine contains details of reconcileEngine
+type reconcileEngine struct {
+	r         *ReconcileChaosEngine
+	reqLogger logr.Logger
+}
+
+//podEngineRunner contains the information of pod
+type podEngineRunner struct {
+	pod, engineRunner *v1.Pod
+	*reconcileEngine
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/types.go]: Can you defile the variable **r** for **reconcileEngine**

File: pkg/controller/chaosengine/types.go [hunk 3]
```diff
@@ -51,3 +53,21 @@ type applicationInfo struct {
...
+}
+
+//serviceEngineMonitor contains informatiom of service
+type serviceEngineMonitor struct {
+	service, engineMonitor *v1.Service
+	*reconcileEngine
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/types.go]: Can you defile the variable **r** for **reconcileEngine**

**Comment by ksatchit** [comment]
Thanks @amitbhatt818 for the refactor!!

**Pull request merged**